webapp/channels/src/components/flag_message_modal/flag_post_modal.tsx

```diff
@@ -0,0 +1,240 @@
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+
+import PostMessagePreview from 'components/post_view/post_message_preview';
+import type {TextboxElement} from 'components/textbox';
+import AdvancedTextbox from 'components/widgets/advanced_textbox/advanced_textbox';
+
+import type {GlobalState} from 'types/store';
+
+import './flag_post_modal.scss';
+
+const noop = () => {};
+
+type SelectedOption = {
+    value: string;
+    label: string;
+}
+
+type Props = {
+    postId: string;
+    onExited: () => void;
+}
+
+export default function FlagPostModal({postId, onExited}: Props) {
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        dispatch(getContentFlaggingConfig());
+    }, [dispatch]);
+
+    const [comment, setComment] = React.useState<string>('');
+    const [reason, setReason] = React.useState<string>('');
+    const [commentError, setCommentError] = React.useState<string>('');
+    const [reasonError, setReasonError] = React.useState<string>('');
+    const [showCommentPreview, setShowCommentPreview] = React.useState<boolean>(false);
+
+    const label = formatMessage({id: 'flag_message_modal.heading', defaultMessage: 'Flag message'});
+    const subHeading = formatMessage({id: 'flag_message_modal.subheading', defaultMessage: 'Flagged messages will be sent to Content Reviewers for review'});
+    const submitButtonText = formatMessage({id: 'generic.submit', defaultMessage: 'Submit'});
+    const requiredCommentSectionTitle = formatMessage({id: 'flag_message_modal.required_comment.title', defaultMessage: 'Comment (required)'});
+    const optionalCommentSectionTitle = formatMessage({id: 'flag_message_modal.optional_comment.title', defaultMessage: 'Comment (optional)'});
+    const reasonSelectPlaceholder = formatMessage({id: 'flag_message_modal.reason_select.placeholder', defaultMessage: 'Select a reason for flagging'});
+    const commentPlaceholder = formatMessage({id: 'flag_message_modal.comment.placeholder', defaultMessage: 'Describe your concern...'});
+
+    const post = useSelector((state: GlobalState) => getPost(state, postId));
+    const channel = useSelector((state: GlobalState) => getChannel(state, post.channel_id));
+    const currentTeam = useSelector(getCurrentTeam);
+    const contentFlaggingSettings = useSelector(contentFlaggingConfig);
+
+    const reasons = useMemo(() => {
+        if (!contentFlaggingSettings || !contentFlaggingSettings.reasons) {
+            return [];
+        }
+
+        return contentFlaggingSettings.reasons.map((reason) => ({
+            value: reason,
+            label: reason,
+        }));
+    }, [contentFlaggingSettings]);
+
+    const previewMetadata: PostPreviewMetadata = useMemo(() => {
+        return {
+            post,
+            post_id: post.id,
```

Comment:
Out of curiosity, why returning `post_id` separately?

Reply:
Hmm, thats interesting. So `PostPreviewMetadata` can accept both post and post ID, but I see no that just passing post_id is sufficient, so I've removed post now.